design-documents/testing/functional/mftf-entities-deprecation-process.md

```diff
@@ -0,0 +1,182 @@
+## MFTF Test Entities Deprecation Process
+
+### Terminologies
+
+`MFTF` - Magento Functional Testing Framework
+
+`MFTF Test Entities` - Variously typed building blocks which can be used to construct `MFTF Tests`. Types of `MFTF Test Entities` are `Test`, `Action Group`, `Page`, `Section`, `Data`, `Meta Data`, and `Suite`
+
+`MFTF Tests` - Tests written to verify Magento functionality using `MFTF`
+
+`BC` - Backward Compatible
+
+`BIC` - Backward Incompatible
+
+`Deprecation Script` - Script which removes all `MFTF Test Entities` with _deprecated_ annotation from Magento 2 Core code before a Magento Major Release
+
+`MFTF Deprecation Static Check` - Static check script which runs during PR and UR build to make sure no reference of deprecated `MFTF Test Entities` are used in Magento 2 core code
+
+### Overview
+
+Over time with the increasing number of `MFTF Tests`, it becomes necessary for developers to add, remove, or rename `MFTF Test Entities` in existing `MFTF Tests`. Taking [MFTF Test Compatibility Policy][] into consideration, however, not all changes are backward compatible. To deal with BIC changes, MFTF deprecation syntax is introduced and deprecation process is defined in this document. Two version of deprecation syntax are presented to gather feedback from broader audience.
+
+#### Applicable Scope for Deprecation
+
+Besides deprecations at the entity level,  action group argument and section element are independent themselves and can be deprecated without deprecating the entire entity. The following table shows full scope of where deprecation can occur.
+
+|**Entity Types**|**Entity Elements**|**Deprecation Applicable?**|
+|---|---|---|
+|ActionGroup|`<actionGroup>`|**Yes**
+| |`<actionGroup>` `<action>`|No
+| |`<actionGroup>` `<argument>`|**Yes**
```

Comment:
What does it look like if an `argument` if deprecated?

For example, if an action group has an `argument` and 5 different `actions` in that action group reference that `argument`, then how can you deprecated it without also affecting those 5 `actions`?

Reply:
@tomreece Good catch! Not all action group argument deprecation can be completely isolated. I think it's good to just remove it from the list. Fixed.